### PR TITLE
feat: Update to Partner latest version 18.1.3

### PR DESCRIPTION
**Pull request by Mansi-mParticle**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
 - Update Urbanairship kit to version 18.1.3 (latest version)
 - Replace deprecated method 'isDataCollectionEnabled' with PrivacyManager.Feature.ALL.
 - Replace 'addEvent' method with recordCustomEvent due to restricted access in the latest version.
 - Updated 'requestRegistrationUpdate' method to match new signature;now passing MParticlePushProvider instance and instanceId (token)
 - Formatted Kotlin code using Kotlin formatter

 ## Testing Plan
 - [x] Was this tested locally? If not, explain why.
 - Ran unit tests and verified functionality with sample application

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-6577

### CODE REVIEW

File: src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt
Comment (BrandonStalnaker) [inline src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt:93]: some formatting changes here that could be unintended? Not sure what best practices are for Kotlin.

**Review by BrandonStalnaker — APPROVED** [review]
Logic looks fine to me. Not as sure about some of the formatting changes but I'll leave that to your expertise.

File: src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt
Comment (Mansi-mParticle) [reply src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt:93]: Automatic corrections made by running the Kotlin formatter

Comment (einsteinx2) [reply src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt:93]: @Mansi-mParticle that's fine, but going forward please make sure to commit formatting changes as a separate commit rather than mixed in with the other changes. It adds significant review overhead as it makes it harder to see which changes are relevant to the PR.

File: src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt
Comment (einsteinx2) [inline src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt:104]: Are you sure that "System Default" is the only potential string option here? I see that we are choosing a default color in the case of this specific value, but in other cases of parse failure (such as a hypothetical "User Default" or something else) we use no color at all? Does that cause visibility problems? Should we default to the system default also any time `Color.parseColor()` fails?

File: src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt
Comment (einsteinx2) [inline src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt:66]: Is this logic backwards? If I'm reading this correctly, we enable all features if they opt out and no features if they opt in?

**Review by einsteinx2 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt
Comment (Mansi-mParticle) [reply src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt:93]: Yes, I can do that. My previous PR failed due to Kotlin lint issues, so now I run the Kotlin formatter command before creating a PR. Do we still need to make a separate PR?

File: src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt
Comment (Mansi-mParticle) [reply src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt:104]: By default, the current code uses a transparent color. I'll also add a fallback option.

File: src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt
Comment (Mansi-mParticle) [reply src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt:66]: Yes, so before, it was working internally like this:

 public void setDataCollectionEnabled(boolean enabled) {
        if (enabled) {
            this.privacyManager.setEnabledFeatures(PrivacyManager.FEATURE_ALL);
        } else {
            this.privacyManager.setEnabledFeatures(PrivacyManager.FEATURE_NONE);
        }
    }

Comment (einsteinx2) [reply src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt:66]: Thanks for the explanation!

File: src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt
Comment (rmi22186) [inline src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt:118]: Seeing these 3 lines duplicated with lines 105-107 is an opportunity for DRYing up.

File: src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt
Comment (Mansi-mParticle) [reply src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt]: this is  client's comment `we default to NotificationCompat.COLOR_DEFAULT, which is Color.TRANSPARENT . the NotificationCompat docs say that means that the system won't decorate the notification with a custom color and will use the system default colors when presenting the notification`. so, mParticle is currently using a default transparent color, so we don't need to handle this anymore.

File: src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt
Comment (Mansi-mParticle) [reply src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt]: Since the issue is the same as mentioned above, removing this code.

File: src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt
Comment (rmi22186) [reply src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt:66]: I don't fully follow @Mansi-mParticle 's comment.  Where was this code "before" and where "internally"?

Comment (Mansi-mParticle) [reply src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt:66]: This code was from Airship version 17.7.1 and is related to the internal Airship class.

File: build.gradle
Comment (rlepinski) [inline build.gradle:44]: We should have a release today that fixes some issues with our IAA products, ideally you can hold off for that release

File: src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt
Comment (rlepinski) [reply src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt:66]: This feels backwards to me as well, if you are opted out you should disable all features?  The above example you use enable but this is the inverse of it so I think it should be switched?

Comment (rlepinski) [reply src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt:66]: Before:
`UAirship.shared().isDataCollectionEnabled = !optedOut`
- false: data collection enabled
- true: data collection disabled

After:
`if (optedOut) PrivacyManager.Feature.ALL else PrivacyManager.Feature.NONE`
- false: data collection disabled (none enabled))
- true: data collection enabled (all enabled)

File: build.gradle
Comment (Mansi-mParticle) [reply build.gradle:44]: sounds good, please let me know once your release is done. and I will address your other comments thanks!

Comment (rmi22186) [reply build.gradle:44]: looks like the urban airship team has updated this to 18.1.3 - https://github.com/urbanairship/android-library/releases/tag/18.1.3

@rlepinski - I assume that @Mansi-mParticle can go ahead and complete this release once you give the thumbs up for her changes?

Comment (rlepinski) [reply build.gradle:44]: Yes, 18.1.3

Comment (Mansi-mParticle) [reply build.gradle]: updated to version 18.1.3 @rlepinski  could you please review the other changes too?

File: src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt
Comment (Mansi-mParticle) [reply src/main/kotlin/com/mparticle/kits/UrbanAirshipKit.kt]: @rlepinski  Can you please confirm these changes?

**Review by rlepinski — APPROVED** [review]
The changes look good to me but I do not have a good way to actually test them.

File: src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt
Comment (einsteinx2) [reply src/main/kotlin/com/mparticle/kits/MParticleAutopilot.kt:93]: @Mansi-mParticle it can be the same PR if it's needed to meet PR check requirements, but should be a separate commit so it's easier to filter out

**Review by einsteinx2 — APPROVED** [review]
lgtm, my previous questions/concerns addressed

**Pull request merged**
